src/main/io/flashfs.c

```diff
@@ -499,7 +501,7 @@ int flashfsIdentifyStartOfFreeSpace(void)
         /* We can choose whatever power of 2 size we like, which determines how much wastage of free space we'll have
          * at the end of the last written data. But smaller blocksizes will require more searching.
          */
-        FREE_BLOCK_SIZE = 2048,
+        FREE_BLOCK_SIZE = 2048, // XXX This can't be smaller than page size for underlying flash device.
```

Comment:
Can we use a `STATIC_ASSERT()` to check for this constraint?

Reply:
As page size changes between NOR and NAND, `STATIC_ASSERT` doesn't work too well here.

Reply:
Hmm, could we put the asserts into the NOR / NAND drivers? Will require pushing down of `FREE_BLOCK_SIZE` into the driver layer though...

Reply:
Defined `FLASH_MAX_PAGE_SIZE` (as 2048) in `drivers/flash.h`. Device page sizes and the allocation size are checked against this value